src/provider/application.js

```diff
@@ -72,7 +82,21 @@ class Application extends EventEmitter {
       const width = calculateWidth(this.resizeConfig.WidthCalculationMethod);
       this.JSONRPC.notification('resize', [null, `${width}px`]);
     } else {
-      const height = calculateHeight(this.resizeConfig.heightCalculationMethod);
+      let height = calculateHeight(this.resizeConfig.heightCalculationMethod);
+
+      // If targetSelectors is specified from Provider or Consumer or both,
+      // need to calculate the height based on specified target selectors
+      if (this.targetSelectors || this.resizeConfig.targetSelectors) {
+        // Combines target selectors from two sources
+        const targetSelectors = [this.targetSelectors, this.resizeConfig.targetSelectors]
+          .filter(val => val)
+          .join(', ');
+
+        height = [].slice.call(document.querySelectorAll(targetSelectors))
+          .map(getOffsetHeightToBody)
+          .reduce((max, curr) => Math.max(max, curr), height);
```

Comment:
its probably more performant to do a single `Math.max` call.
```js
const heights = [].slice.call(document.querySelectorAll(targetSelectors)).map(getOffsetHeightToBody);
heights.push(bodyOffsetHeight);
const maxHeight = Math.max(...heights);
```

Reply:
updated in the latest commit.